src/lib_ccx/output.c

```diff
@@ -50,6 +50,7 @@ int init_write(struct ccx_s_write *wb, char *filename, int with_semaphore)
 	wb->fh = -1;
 	wb->temporarily_closed = 0;
 	wb->filename = filename;
+	wb->original_filename = strdup(filename);
```

Comment:
This should be freed in dinit_write, otherwise we have a leak.
Other than that this PR LGTM. Can you correct that and I'll merge?

Reply:
Ah, thanks for catching that. Fixed.